### ISSUE TITLE
[CoE Starter Kit - BUG] COE keeps asking us to republish apps

### ISSUE THREAD

**Issue by Service-WebDevPowerP** [microsoft/coe-starter-kit#9017]
### Does this bug already exist in our backlog?

- [X] I have checked and confirm this is a new bug.

### Describe the issue

We recevie emails saying "Keep your app updated to stay compliant.
You own an app that's currently shared or used broadly but which has not been published recently"

I have republished each app every week for at least a month now and we still receive the email each week saying it needs to be republished. Is this expected behaviour? I thought this was only meant to trigger every 90 days?

we have some flows that have turned off since August update, could that be the cause?

![image](https://github.com/user-attachments/assets/1478cafb-4be8-4be2-8364-f1875ebf9603)

I am not sure which solution contains this flow so I am not sure which version it is, but we currently have Septembers update installed 

### What solution are you experiencing the issue with?

Audit Log

### What solution version are you using?

September

### What app or flow are you having the issue with?

Unknown

### What method are you using to get inventory and telemetry?

None

**Comment by Jenefer-Monroe** [comment]
Your inventory generally will not run correctly without these 3 flows turned on. As you can see the driver flow is off so you have not gathered inventory since August 20th. 
![image](https://github.com/user-attachments/assets/8bc71dda-2ddd-4a85-a71d-b2327a984351)

**Comment by Jenefer-Monroe** [comment]
Please do boot the setup wizard and walk through initial setup it will ensure the first two are turned on and running. 
Then manually turn on Admin | Audit Logs | Update Data (V2)

## Use the setup wizard
Its highly recommended that you use the Setup Wizard to walk you through setup and upgrades. It will ensure all the new flows are turned on in the correct order, trigger things like population of the bookmarks screen and do other cleanup steps needed.
You should do this both for upgrades and clean installs.
To use it, first perform the import of the solution install or upgrade manually, then boot the app and walk through the screens.
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/b84cc1ff-383a-497c-9236-569e8313c73c)
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/ed02f0d0-c276-48cc-90af-47e5bf5a928e)

**Comment by Jenefer-Monroe** [comment]
For the publish warning, this is in the following flow: **_Admin | Compliance detail request v3_**
And is driven by the environment variable below. Its possible that the inventory not running is what happened, because if the app has never been published it will also go through the process. 

![image](https://github.com/user-attachments/assets/72940412-7282-4e8d-9130-6ce20be9cb3a)
![image](https://github.com/user-attachments/assets/80cfd6ee-e1e0-4eaa-b62f-5d774ca385b3)

**Comment by Service-WebDevPowerP** [comment]
Hi Jenefer

Thank you for your reply. I've always done the upgrades through the Set up and Upgrade wizard. Any idea why it didn't turn that flow on? Our Test environement has done the same thing. I have turned it on for both environements. Should I run the flow once I have turned it on?

**Comment by Jenefer-Monroe** [comment]
Yes it looks like you may not have updated the DLP policy with the new connector

Please add the **_DLP Policy - Power Platform Admin V2_** connector to your DLP. Then once it has had time to sync up (which can take a while unfortunately) turn on the following flows in the order shown:

- HELPER - Driver Escalation Check
- Admin | Sync Template v4 (Driver)

**Comment by Service-WebDevPowerP** [comment]
Thank you Jenefer, I will let you know how that goes

**Comment by Service-WebDevPowerP** [comment]
HELPER - Driver Escalation Check doesnt appear to be in any of the solutions. Where is it?

**Comment by Service-WebDevPowerP** [comment]
![image](https://github.com/user-attachments/assets/a644ebb1-0d3d-4374-ac84-ee3077b05147)

Hope this helps, no not a sovereign  cloud user

**Comment by Jenefer-Monroe** [comment]
Ok that flow is in the core solution then. It appears there may be something wrong going on. 
Please go to the solution itself, outside of the command center, run the following flow:
![image](https://github.com/user-attachments/assets/dde3ebe1-0526-4a38-b0b9-336f727a6690)

Then see if you can find the escalation helper in the list an turn it on
![image](https://github.com/user-attachments/assets/83307b57-4e1e-4897-9ec3-9f26ffe7be56)

**Comment by Service-WebDevPowerP** [comment]
Thank you I have now found them both and turned them on. Anything else I need to do?

**Comment by Jenefer-Monroe** [comment]
You will need to make sure all the flows in inventory are turned on. 

Please go to the Admin Command Center > CoE Flows > **Inventory**. Validate that all flows are turned on and that there are no unmanaged layers as shown here. 
![image](https://github.com/microsoft/coe-starter-kit/assets/89584709/598b5024-2e1a-4390-a6b2-49644368c246)

**Comment by Service-WebDevPowerP** [comment]
We had 2 flows that were off they are now on. Strangely the Admin command center was out of sync with the state of the flows. Saying some were off when they were on, it has since refreshed though. Anything else I need to do?

**Comment by Jenefer-Monroe** [comment]
I presume that since inventory has been off for about 5 weeks, we need to do something more to get you back up to date.
I suggest updating the **_InventoryFilter_DaysToLookBack_** environment variable to go backwards further, 45 days should do it (see notes below)
Then when that is complete see if your publish date is correct.

## Change how far inventory does lookback 
In order to avoid using unnecessary API calls the kit only looks back 7 days (by default) to look for updated flows. 
In the event something happened and your inventory was off for a longer period of time than that, we do allow you do change this and look back further. Here is how to do that. 

1. Update the env variable to go back to when your inventory stopped working: **_InventoryFilter_DaysToLookBack_**
![image](https://github.com/user-attachments/assets/7669b70c-76dd-444f-bc43-d3f5712ac86c)
2. Restart all the flows that use this env variable to ensure the flows do not cache the old value
![image](https://github.com/user-attachments/assets/4dad5818-49db-4ce4-9d74-863bb81bfa91)
![image](https://github.com/user-attachments/assets/8bbe3721-f841-45a1-9e06-40b5ffa1d647)
3. Kick off your driver flow: **_Admin | Sync Template v4 (Driver)_** and wait for it to complete
4. Reset the environment variable back to 7 to avoid needlessly consuming these API calls
5. Restart all those flows again.

**Comment by Jenefer-Monroe** [comment]
closing out as no further action for starter kit team
